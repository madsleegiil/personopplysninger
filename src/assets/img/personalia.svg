<?xml version="1.0" encoding="UTF-8"?>
<svg width="52px" height="62px" viewBox="0 0 52 62" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
    <title>book-contact-2 Copy</title>
    <desc>Created with Sketch.</desc>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="book-contact-2-Copy">
            <path d="M6.74545455,0 L49.1454545,0 L49.1454545,12.7179487 L6.74545455,12.7179487 C3.19744342,12.7179487 0.321212121,9.87093892 0.321212121,6.35897436 C0.321212121,2.84700979 3.19744342,6.45137425e-16 6.74545455,0 Z" id="Rectangle" fill="#E7E9E9"></path>
            <path d="M50.4303032,10.1743589 L6.74545447,10.1743589 C4.62031506,10.1743589 2.89090899,8.46252298 2.89090899,6.35897426 C2.89090899,4.25542553 4.62031506,2.54358963 6.74545447,2.54358963 L50.4303032,2.54358963 C51.1395396,2.54358963 51.7151517,1.97382553 51.7151517,1.27179475 C51.7151517,0.572307567 51.1395396,-1.27999027e-07 50.4303032,-1.27999027e-07 L6.74545447,-1.27999027e-07 C3.20184233,-1.27999027e-07 0.321212009,2.85390758 0.321212009,6.35897426 C0.321212009,6.65911785 0.341769585,6.95417426 0.380315039,7.2415999 C0.341769585,7.3662358 0.321212009,7.49595888 0.321212009,7.63076913 L0.321212009,54.6871796 C0.321212009,58.1922463 3.20184233,61.046154 6.74545447,61.046154 L50.4303032,61.046154 C51.1395396,61.046154 51.7151517,60.4763899 51.7151517,59.7743591 L51.7151517,11.4461538 C51.7151517,10.7466666 51.1395396,10.1743589 50.4303032,10.1743589 Z" id="Fill/book-contact-2" fill="#D6897D" fill-rule="nonzero"></path>
            <path d="M37.5818183,48.3282051 L14.4545455,48.3282051 C13.7453091,48.3282051 13.169697,47.758441 13.169697,47.0564103 C13.169697,39.3010051 17.2966303,35.5542974 21.2976485,34.0001641 C19.4834424,32.6037333 18.3090909,30.4264205 18.3090909,27.9794872 C18.3090909,23.7723897 21.767903,20.3487179 26.0181818,20.3487179 C30.2684606,20.3487179 33.7272729,23.7723897 33.7272729,27.9794872 C33.7272729,30.4264205 32.5529214,32.6037333 30.7387152,34.0001641 C34.7397335,35.5517538 38.8666668,39.3010051 38.8666668,47.0564103 C38.8666668,47.758441 38.2910547,48.3282051 37.5818183,48.3282051 Z" id="Path" fill="#E7E9E9" fill-rule="nonzero"></path>
            <path d="M6.74545447,5.08717938 C6.0362181,5.08717938 5.46060598,5.65948707 5.46060598,6.35897426 C5.46060598,7.06100503 6.0362181,7.63076914 6.74545447,7.63076914 L47.8606062,7.63076914 C48.5698426,7.63076914 49.1454547,7.06100503 49.1454547,6.35897426 C49.1454547,5.65948707 48.5698426,5.08717938 47.8606062,5.08717938 L6.74545447,5.08717938 Z" id="Path" fill="#C6C2BF"></path>
        </g>
    </g>
</svg>