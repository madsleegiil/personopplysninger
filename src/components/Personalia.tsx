import * as React from 'react';

const Personalia = () => {
    return (
        <div className="panel">
            Personalia
        </div>
    );
};

export default Personalia;
