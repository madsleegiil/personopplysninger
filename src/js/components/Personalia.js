import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Box from 'js/components/Box';

class Personalia extends Component {
  render() {
    return (
      <Box header="Personalia">
        <ul className="personalia-list">
          <li>
            <span className="title">Fornavn</span>
            <span className="content">{this.props.navn.fornavn}</span>
          </li>
          <li>
            <span className="title">Etternavn</span>
            <span className="content">{this.props.navn.slektsnavn}</span>
          </li>
          <li>
            <span className="title">Fødselsnummer</span>
            <span className="content">{this.props.ident}</span>
          </li>
          <li>
            <span className="title">Utenlandsk ID</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Telefon</span>
            <span className="content">{this.props.telefon.privat}</span>
          </li>
          <li>
            <span className="title">Kontonummer</span>
            <span className="content">{this.props.kontonummer.nummer}</span>
          </li>
          <li>
            <span className="title">e-post</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Språk</span>
            <span className="content">{this.props.spraak.kode}</span>
          </li>
          <li>
            <span className="title">Statsborgerskap</span>
            <span className="content">{this.props.statsborgerskap.kode}</span>
          </li>
          <li>
            <span className="title">Personstatus</span>
            <span className="content">{this.props.status.kode}</span>
          </li>
          <li>
            <span className="title">Fødested</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Spesialregistrering</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Sivilstand</span>
            <span className="content">{this.props.sivilstand.kode}</span>
          </li>
          <li>
            <span className="title">Sikkerhetstiltak</span>
            <span className="content">{this.props.tiltak.type}</span>
          </li>
          <li>
            <span className="title">Kjønn</span>
            <span className="content">{this.props.kjonn}</span>
          </li>
          <li>
            <span className="title">Tilpasset kommunikasjon</span>
            <span className="content">x</span>
          </li>
        </ul>
        <div className="box-footer">
          Kilde: NAV, Skatteetaten, Folkeregisteret, Kontakt- og reservasjonsregisteret
        </div>
      </Box>
    );
  }
}

Personalia.propTypes = {
  navn: PropTypes.shape.isRequired,
  ident: PropTypes.string.isRequired,
  statsborgerskap: PropTypes.shape.isRequired,
  status: PropTypes.shape.isRequired,
  telefon: PropTypes.shape.isRequired,
  tiltak: PropTypes.shape.isRequired,
  kjonn: PropTypes.string.isRequired,
  spraak: PropTypes.shape.isRequired,
  sivilstand: PropTypes.shape.isRequired,
  kontonummer: PropTypes.shape.isRequired,
};

export default Personalia;
