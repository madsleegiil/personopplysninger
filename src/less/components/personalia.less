.personalia-list {
  list-style: none;
  columns: 2;

  li {
    padding: 1rem;

    .title {
      font-weight: bold;
      display: block;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }
  }
}
