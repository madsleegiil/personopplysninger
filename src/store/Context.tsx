import React, { createContext, useContext, useReducer, Dispatch } from "react";
import { Action, Locale, reducer, Store } from "./Store";
import { FetchNameInfo } from "./providers/Auth";
import { FetchFeatureToggles } from "./providers/FeatureToggles";
import { FetchDsopInfo } from "../pages/digital-samhandling-offentlig-privat/DsopFetch";
import { FetchInstInfo } from "../pages/institusjonsopphold/InstFetch";
import { FetchPersonInfo } from "./providers/PersonInfo";
import { FetchKontaktInfo } from "../pages/forside/sections/4-personinfo/2-kontaktinfo/subsections/kontakt-og-reservasjonsregisteret/DKIF-Fetch";
import { Fetchskattetrekksmeldinger } from "../pages/skattetrekksmelding/SkattFetch";
import { FetchMedlInfo } from "../pages/medlemskap-i-folketrygden/MedlFetch";

interface Props {
  children: JSX.Element | JSX.Element[];
}

export const StoreContext = createContext({} as [Store, Dispatch<Action>]);
export const StoreProvider = (props: Props) => {
  const { children } = props;

  const initialLocale = (window.location.pathname.includes("/en")
    ? "en"
    : "nb") as Locale;

  const initialState = {
    locale: initialLocale,
    nameInfo: { status: "LOADING" } as FetchNameInfo,
    featureToggles: {
      status: "LOADING",
      data: {
        "personopplysninger.pdl": false,
        "personopplysninger.dsop": false,
        "personopplysninger.inst": false,
        "personopplysninger.skatt": false,
        "personopplysninger.medl": false,
        "personopplysninger.fullmakt": false,
        "personopplysninger.tilrettelegging": false,
        "pdl-fullmakt": false,
      },
    } as FetchFeatureToggles,
    dsopInfo: { status: "LOADING" } as FetchDsopInfo,
    instInfo: { status: "LOADING" } as FetchInstInfo,
    personInfo: { status: "LOADING" } as FetchPersonInfo,
    kontaktInfo: { status: "LOADING" } as FetchKontaktInfo,
    skattetrekksmeldinger: { status: "LOADING" } as Fetchskattetrekksmeldinger,
    medlInfo: { status: "LOADING" } as FetchMedlInfo,
  };

  return (
    <StoreContext.Provider value={useReducer(reducer, initialState)}>
      {children}
    </StoreContext.Provider>
  );
};
export const useStore = () => useContext(StoreContext);
