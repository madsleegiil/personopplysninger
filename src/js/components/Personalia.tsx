/* eslint-disable */

import React, { Component } from "react";
import { FormattedMessage } from "react-intl";
import { Undertekst } from "nav-frontend-typografi";
import Box from "./Box";
import kvinne from "../../assets/img/kvinne.svg";
import mann from "../../assets/img/mann.png";
import ListElement from "./ListElement";
import { Personalia as PersonaliaType } from "../../types/personalia";

class Personalia extends Component<PersonaliaType> {
  render() {
    const {
      personident,
      fornavn,
      etternavn,
      kontonr,
      spraak,
      statsborgerskap,
      foedested,
      sivilstand,
      kjoenn
    } = this.props;

    const personidentHeader =
      personident && personident.type === "DNR"
        ? "personalia.dnr"
        : "personalia.fnr";

    const fornavnHeader =
      fornavn && fornavn.indexOf(" ") === -1
        ? "personalia.first_name"
        : "personalia.first_and_middle_name";

    return (
      <Box
        id="personalia"
        header="Personalia"
        icon={kjoenn === "Mann" ? mann : kvinne}
        infoType="personalia"
      >
        <ul className="list-column-2">
          {fornavn && <ListElement titleId={fornavnHeader} content={fornavn} />}
          {etternavn && (
            <ListElement
              classNameContent="capitalize"
              titleId="personalia.surname"
              content={etternavn}
            />
          )}
          {personident && personident.verdi && (
            <ListElement
              titleId={personidentHeader}
              content={personident.verdi}
            />
          )}
          {kontonr && (
            <ListElement titleId="personalia.account_no" content={kontonr} />
          )}
          {spraak && (
            <ListElement titleId="personalia.language" content={spraak} />
          )}
          {statsborgerskap && (
            <ListElement
              titleId="personalia.citizenship"
              content={statsborgerskap}
            />
          )}
          {foedested && (
            <ListElement titleId="personalia.birthplace" content={foedested} />
          )}
          {sivilstand && (
            <ListElement
              titleId="personalia.civil_status"
              content={sivilstand}
            />
          )}
          {kjoenn && (
            <ListElement titleId="personalia.gender" content={kjoenn} />
          )}
        </ul>
        <div className="box-footer">
          <Undertekst>
            <FormattedMessage id="personalia.source" />
          </Undertekst>
        </div>
      </Box>
    );
  }
}

export default Personalia;
