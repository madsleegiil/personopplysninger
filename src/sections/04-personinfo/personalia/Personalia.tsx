/* eslint-disable */

import React, { Component } from "react";
import Box from "../../../components/box/Box";
import Kilde from "../../../components/kilde/Kilde";
import kvinne from "../../../assets/img/kvinne.svg";
import mann from "../../../assets/img/mann.png";
import ListElement from "../../../components/listelement/ListElement";
import { Personalia as PersonaliaType } from "../../../types/personalia";

interface Props {
  personalia: PersonaliaType;
}

class Personalia extends Component<Props> {
  render() {
    const {
      personident,
      fornavn,
      etternavn,
      kontonr,
      spraak,
      statsborgerskap,
      foedested,
      sivilstand,
      kjoenn
    } = this.props.personalia;

    const personidentHeader =
      personident && personident.type === "DNR"
        ? "personalia.dnr"
        : "personalia.fnr";

    const fornavnHeader =
      fornavn && fornavn.indexOf(" ") === -1
        ? "personalia.first_name"
        : "personalia.first_and_middle_name";

    const formattertKontonr =
      kontonr && kontonr.length === 11
        ? kontonr.replace(/^(.{4})(.{2})(.*)$/, "$1 $2 $3")
        : kontonr;

    const formattertPersonident =
      personident && personident.verdi.length === 11
        ? {
            ...personident,
            verdi: personident.verdi.replace(/^(.{6})(.*)$/, "$1 $2")
          }
        : personident;

    return (
      <Box
        id="personalia"
        header="Personalia"
        icon={kjoenn === "Mann" ? mann : kvinne}
        infoType="personalia"
      >
        <ul className="list-column-2">
          <ListElement titleId={fornavnHeader} content={fornavn} />
          <ListElement
            className="capitalize"
            titleId="personalia.surname"
            content={etternavn}
          />
          {formattertPersonident && (
            <ListElement
              titleId={personidentHeader}
              content={formattertPersonident.verdi}
            />
          )}
          <ListElement titleId="personalia.language" content={spraak} />
          <ListElement
            titleId="personalia.citizenship"
            content={statsborgerskap}
          />
          <ListElement titleId="personalia.birthplace" content={foedested} />
          <ListElement titleId="personalia.civil_status" content={sivilstand} />
          <ListElement titleId="personalia.gender" content={kjoenn} />
        </ul>
        <Kilde
          tekst="personalia.source"
          lenkeTekst="personalia.link.folkeregisteret"
          href="https://www.skatteetaten.no/person/folkeregister/"
        />
        <hr className="box__linje-bred" />
        <ul className="list-column-2">
          <ListElement
            titleId="personalia.account_no"
            content={formattertKontonr}
          />
        </ul>
      </Box>
    );
  }
}

export default Personalia;
