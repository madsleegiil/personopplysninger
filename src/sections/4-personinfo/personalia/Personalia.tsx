import React, { Component } from "react";
import Box from "../../../components/box/Box";
import kvinne from "../../../assets/img/kvinne.svg";
import mann from "../../../assets/img/mann.png";
import Kontonummer from "./subsections/Kontonummer";
import Folkeregisteret from "./subsections/Folkeregisteret";
import Telefonnummer from "./subsections/Telefonnummer";
import DKIF from "./subsections/DKIF";
import { Personalia as PersonaliaType } from "../../../types/personalia";

interface Props {
  personalia: PersonaliaType;
}

class Personalia extends Component<Props> {
  render() {
    const { kontonr, kjoenn } = this.props.personalia;

    return (
      <Box
        id="personalia"
        header="Personalia"
        icon={kjoenn === "Mann" ? mann : kvinne}
        infoType="personalia"
      >
        <hr className="box__linje-bred" />
        <Folkeregisteret personalia={this.props.personalia} />
        <Kontonummer kontonummer={kontonr} />
        <Telefonnummer tlfnr={this.props.personalia.tlfnr} />
        <DKIF />
      </Box>
    );
  }
}

export default Personalia;
