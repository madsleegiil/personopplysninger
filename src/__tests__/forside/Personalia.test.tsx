import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import personInfo from "../../clients/apiMock/app/fetch/person-info.json";
import Personalia from "../../pages/forside/sections/4-personinfo/1-personalia/Personalia";
import { Personalia as PersonaliaType } from "types/personalia";
import { StoreProvider } from "../../store/Context";

jest.mock("react-modal");

describe("Personalia", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <StoreProvider>
          <IntlProvider locale={"nb"} messages={nbMessages}>
            <Personalia personalia={personInfo as unknown as PersonaliaType} />
          </IntlProvider>
        </StoreProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
