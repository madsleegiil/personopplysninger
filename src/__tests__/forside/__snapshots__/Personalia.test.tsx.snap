// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personalia renders correctly 1`] = `
<DocumentFragment>
  <div
    class="box__wrapper"
    id="personalia"
  >
    <div
      class="navds-guide-panel navds-guide-panel--poster"
    >
      <div
        class="navds-guide"
      >
        <div
          class="navds-guide__illustration navds-guide__illustration--medium"
        >
          <img
            alt="Veileder"
            class="box__ikon"
            src="Personalia.svg"
          />
        </div>
      </div>
      <div
        class="navds-guide-panel__content"
      >
        <div
          class="box__container"
        >
          <div
            class="box__header"
          >
            <div
              class="box__title-container"
            >
              <div
                class="box__line"
              />
              <h2
                class="box__title navds-heading navds-heading--medium"
              >
                Personalia
              </h2>
              <button
                aria-label="Informasjonstekst"
                aria-pressed="false"
                class="infotekst__title-i-button"
                title="Informasjon"
              >
                <img
                  alt="Vis mer informasjon"
                  class="infotekst__title-i-icon"
                  src="Informasjonstekst.svg"
                />
              </button>
              <div
                class="box__line"
              />
            </div>
            <a
              class="navds-link anchor-link"
              href="#personalia"
            >
              <img
                alt=""
                src="AnchorLink.svg"
              />
              Lenke hit
            </a>
          </div>
          <div
            class="box__content"
          >
            <ul
              class="list-column-2"
            />
            <div
              class="kilde__container"
            >
              <p
                class="navds-body-long"
              >
                <a
                  class="navds-link kilde__lenke lenke"
                  href="https://www.skatteetaten.no/person/folkeregister/"
                >
                  <span
                    class="kilde__icon"
                  >
                    <img
                      alt="Ekstern lenke"
                      src="Link.svg"
                    />
                  </span>
                  Endre i Folkeregisteret
                </a>
              </p>
              <div
                class="kilde__tekst"
              >
                <p
                  class="navds-detail"
                >
                  Kilde: Folkeregisteret
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
